.github/workflows/CI.yml

```diff
@@ -75,4 +139,165 @@ jobs:
+          echo "$(go env GOPATH)/bin" >> $GITHUB_PATH
+          echo "TOKEN_PRIVATE_KEY_PATH=${GITHUB_WORKSPACE}/src/github.com/goharbor/harbor-arm/tests/private_key.pem" >> $GITHUB_ENV
+          IP=`hostname -I | awk '{print $1}'`
+          echo "IP=$IP" >> $GITHUB_ENV
+        shell: bash
       
+      - name: install
+        run: |
+          cd src/github.com/goharbor/harbor-arm/src/github.com/goharbor/harbor
+          env
+          df -h
+          bash ./tests/showtime.sh ./tests/ci/api_common_install.sh $IP LDAP
+      - name: script
+        run: |
+          echo IP: $IP
+          df -h
+          cd src/github.com/goharbor/harbor-arm/src/github.com/goharbor/harbor
+          bash ./tests/showtime.sh ./tests/ci/api_run.sh LDAP $IP
+          df -h
+  
+  OFFLINE:
+    env:
+      OFFLINE: true
+    runs-on: [self-hosted, linux, ARM64, ci]
+      #- self-hosted
+    timeout-minutes: 100
+    steps:
+      - name: Set up Go 1.15
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.15.12
+        id: go
+      - uses: actions/checkout@v2
+        with:
+          path: src/github.com/goharbor/harbor-arm
+      - name: setup env
+        run: |
+          cd src/github.com/goharbor/harbor-arm
+          sudo make download compile_redis prepare_arm_data pre_update
+          pwd
+          docker version
+          go env
+          echo "GOPATH=$(go env GOPATH):$GITHUB_WORKSPACE" >> $GITHUB_ENV
+          echo "$(go env GOPATH)/bin" >> $GITHUB_PATH
+          echo "TOKEN_PRIVATE_KEY_PATH=${GITHUB_WORKSPACE}/src/github.com/goharbor/harbor-arm/tests/private_key.pem" >> $GITHUB_ENV
+        shell: bash
+      - name: before_install
+        run: |
+          set -x
+          cd src/github.com/goharbor/harbor-arm
+          pwd
+          env
+          df -h
+          #sudo apt install -y xvfb
+          #xvfb-run ls
+          IP=`hostname -I | awk '{print $1}'`
+          echo '{"insecure-registries" : ["'$IP':5000"]}' | sudo tee /etc/docker/daemon.json
+          echo "IP=$IP" >> $GITHUB_ENV
+          sudo cp ./tests/harbor_ca.crt /usr/local/share/ca-certificates/
+          sudo update-ca-certificates
+          sudo service docker restart
+      - name: script
+        run: |
+          echo IP: $IP
+          df -h
+          cd src/github.com/goharbor/harbor-arm/src/github.com/goharbor/harbor
+          bash ./tests/showtime.sh ./tests/ci/distro_installer.sh
+          df -h
+  
+  UI_UT:
+    env:
+      UI_UT: true
+    runs-on: [self-hosted, linux, ARM64, ci]
+      #- self-hosted
+    timeout-minutes: 100
+    steps:
+      - uses: actions/setup-node@v1
+        with:
+          node-version: '10.16.2'
+      - uses: actions/checkout@v2
+        with:
+          path: src/github.com/goharbor/harbor-arm
+      - name: script
+        run: |
+          echo IP: $IP
+          df -h
+          cd src/github.com/goharbor/harbor-arm
+          sudo make download compile_redis prepare_arm_data pre_update
```

Comment:
could remove this line

Reply:
This line cannot be removed because the harbor source code needs to be downloaded